airflow-core/src/airflow/models/dagbag.py

```
@@ -147,12 +147,44 @@ def handle_timeout(signum, frame):
 
 
 def _validate_executor_fields(dag: DAG) -> None:
+    """Validate that executors specified in tasks are available and owned by the same team as the dag bundle."""
+    import logging
+
+    log = logging.getLogger(__name__)
+    dag_team_name = None
+
+    # Get team name from bundle configuration if available
+    if hasattr(dag, "bundle_name") and dag.bundle_name:
+        try:
+            from airflow.dag_processing.bundles.manager import DagBundlesManager
+
+            bundle_manager = DagBundlesManager()
+            bundle_config = bundle_manager._bundle_config[dag.bundle_name]
+            # TODO[multi-team] Raise exceptions below instead of logging once we have a multi-team feature flag configuration
+            dag_team_name = bundle_config.team_name
+            log.debug(
+                "Found team '%s' for DAG '%s' via bundle '%s'", dag_team_name, dag.dag_id, dag.bundle_name
+            )
+        except Exception as e:
+            log.debug(
```

I am trying to understand in which scenario we end up in this branch. Can you provide an example? My feeling tells me this is a anormal situation, thus we should raise an exception, not swallow it

---

Yupp, see the ToDo above. For now I do not want this code to fail the dag parsing in ANY possible way, so all exceptions are caught and logged. As we test, tweak and gain comfort with this approach this debug statement will be changed to an exception being raised (or the try/catch removed and the exceptions allowed to bubble up).

---

Sounds good

---

I will likely switch this to exception once we have the multi-team flag from your PR @vincbeck I think that will be a great compromise!